src/esp/assets/managers/ObjectAttributesManager.cpp

```diff
@@ -21,60 +21,85 @@ namespace esp {
 namespace assets {
 
 namespace managers {
-const PhysicsObjectAttributes::ptr
-ObjectAttributesManager::createAttributesTemplate(
+PhysicsObjectAttributes::ptr ObjectAttributesManager::createAttributesTemplate(
     const std::string& attributesTemplateHandle,
     bool registerTemplate) {
   if (assetAttributesMgr_->getTemplateLibHasHandle(attributesTemplateHandle)) {
     // if attributesTemplateHandle == some existing primitive attributes, then
     // this is a primitive-based object we are building
     return createPrimBasedAttributesTemplate(attributesTemplateHandle,
                                              registerTemplate);
-  } else {
-    // if attributesTemplateHandle != some existing primitive attributes, then
+  } else if (Corrade::Utility::Directory::exists(attributesTemplateHandle)) {
+    // if attributesTemplateHandle == some existing file then
     // assume this is a file-based object template we are building.
     return createFileBasedAttributesTemplate(attributesTemplateHandle,
                                              registerTemplate);
+  } else {
+    // if neither of these is true, then build an empty template and assign the
+    // passed handle to its origin handle and its render asset handle
+    return createEmptyAttributesTemplate(attributesTemplateHandle,
```

Comment:
Can you talk about the use cases for this function? Will the user ever intentionally create an empty template, or is that strictly an error-condition fallback? 

This behavior of looking for a file and then falling back to creating an empty template... it seems confusing and error-prone. Doesn't the user want an error/warning if the file she was expecting is missing?

Reply:
Current tests rely on the ability to create a blank object attributes template and then configure it as desired.  Also, it is conceivable that a user would wish to have a blank attributes they could consume to create objects manually, without having to configure physics/scenes to load object json.  Perhaps feedback to the effect of "<type> attributes created" might be helpful, but in this PR we are assuming the user knows what they are doing.

This will be clearly described in tutorials consuming these functions, of course. :)

Reply:
Feedback added.

Reply:
>in this PR we are assuming the user knows what they are doing
I'm really just asking for consistency with the error-handling in the rest of this file. I like the LOG(INFO) you added because it's better than silent failure. But it would be more consistent with other code in this file to issue a LOG(ERROR) in the case of a missing file. And I understand the current interface makes it difficult to distinguish the error-case from the making-a-blank-template case, so I suspect the interface needs to change.

Reply:
Originally I had an error trap here as you describe, but then I realized i needed to be able to support the creation of a blank template - passing an unknown template handle could denote the user wanting a blank one with the given name.  I considered setting some kind of sentinel value for the name to denote that it was intended to be a new template (like "newTemplate") but this seemed suboptimal because it too could fail if it was misspelled or unknown, and it would be too easy for the user to forget to rename the template, causing later template creations to overwrite earlier ones.  

Also, as far as the user goes, I do not think there's going to be an expectation bias toward any type of attributes to be created here.  The file based attributes in a particular scene are usually specified in the scene json, and will be created when the scene loads, so the user won't even access this.  She is just as likely (in fact, i would expect more likely) to be wanting to manually create a prim-based object attributes here as a file based one, and there are certainly use cases for blank attributes as well.

This is why I chose the method I did, the least of evils as I saw it, but i am certainly interested in revisiting this at a later date once folks have consumed it and can give some feedback.